### PR TITLE
fix(ecs): `autoscalingGroup` attribute does not exist on `ImportedCluster`

### LINKED ISSUE #29241: aws-ecs: `Cluster.fromClusterAttributes` does not properly set `autoScalingGroup`
### Describe the bug

`Cluster.fromClusterAttributes` does not properly set `autoScalingGroup`

### Expected Behavior

`autoScalingGroup` to be set

### Current Behavior

Is `undefined` which leads to NPEs.

### Reproduction Steps

```
    const vpc   = new Vpc(this, 'Vpc', {});
    const cluster = Cluster.fromClusterAttributes(this, 'Cluster', {
      clusterName: 'my-cluster',
      vpc,
      securityGroups: [],
      autoscalingGroup: new AutoScalingGroup(this, 'Asg', {
        vpc,
        instanceType: InstanceType.of(InstanceClass.A1, InstanceSize.MICRO),
        machineImage: MachineImage.latestAmazonLinux2(),
      }),
    });
    cluster.autoscalingGroup!.addUserData('yum install -y aws-cli');
```

### Possible Solution

All `ClusterAttribute`s should get set on the `ImportCluster`.

### Additional Information/Context

_No response_

### CDK CLI Version

2.126.0

### Framework Version

_No response_

### Node.js Version

v20.10.0

### OS

OSX

### Language

TypeScript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by msambol**
This PR adds an `autoscalingGroup` attribute to `ImportedCluster`. This attribute is set using the `autoscalingGroup` property from `ClusterAttributes` in the constructor of `ImportedCluster`.

Closes #29241.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by msambol** [comment]
Exemption request: I don't believe integration is required here.

File: packages/aws-cdk-lib/aws-ecs/lib/cluster.ts [hunk 1]
```diff
@@ -858,6 +858,11 @@ class ImportedCluster extends Resource implements ICluster {
    */
   public readonly hasEc2Capacity: boolean;
 
+  /**
+   * Autoscaling group of the cluster
+   */
+  readonly autoscalingGroup?: autoscaling.IAutoScalingGroup;
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-ecs/lib/cluster.ts:864]:
```suggestion
   * The autoscaling group added to the cluster if capacity is associated to the cluster
   */
  public readonly autoscalingGroup?: autoscaling.IAutoScalingGroup;
```
Let's align it with other variables' declarations.

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Looks good overall 👍
It would be nice to add integration test coverage if possible, similar to what we have for [EKS](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk-testing/framework-integ/test/aws-eks/test/integ.eks-cluster-imported.ts).

**Comment by msambol** [comment]
@lpizzinidev Added integration test 👍🏼

**Review by lpizzinidev — APPROVED** [review]
Thanks 👍

**Comment by msambol** [comment]
@Mergifyio refresh

**Comment by msambol** [comment]
@tim-finnigan Can you queue this one up? I think it's stuck.

**Comment by msambol** [comment]
@Mergifyio refresh

**Comment by msambol** [comment]
@Mergifyio refresh

**Pull request merged**